### PR TITLE
feat(cli): allow hiding scripts from the help list

### PR DESCRIPTION
**Pull request by sxn**
**What**: Allow scripts to be hidden from the help list.

**Why**: In order to allow having semi-private scripts - semi because they will still be runnable as long as you know they're there.

**How**: by changing the ```src/resolve-script-object-to-string.js::resolveScriptObjectToScript``` function so that if the ```hiddenFromHelp``` property is defined and truthy, undefined is returned.

### CODE REVIEW

File: src/resolve-script-object-to-string.test.js [hunk 1]
```diff
@@ -32,7 +44,7 @@ test('resolves default to the script if it is a string', t => {
   t.is(result, 'string')
 })
 
-test('does not resolve a script object without a script or defualt', t => {
+test('does not resolve a script object without a script or default', t => {
```
Comment (kentcdodds) [inline src/resolve-script-object-to-string.test.js]: 👍

File: src/resolve-script-object-to-string.js [hunk 2]
```diff
@@ -12,13 +12,22 @@ function resolveScriptObjectToString(script) {
...
-    }
+    return resolvePlainObjectToScript(script)
   } else if (isString(script)) {
     return {script}
   }
   return undefined
 }
+
+function resolvePlainObjectToScript(script) {
+  if (!isUndefined(script.hiddenFromHelp)) {
```
Comment (kentcdodds) [inline src/resolve-script-object-to-string.js]: I think this should probably simply be: `if (script.hiddenFromHelp) {`

I think we're good to go with `truthy`. So `hiddenFromHelp: false` should not hide it from help ;-)

Comment (sxn) [inline src/resolve-script-object-to-string.js]: Huh. I could've sworn that I pushed the truthy-checking version. Obviously I didn't though. :fearful:

**Comment by kentcdodds** [comment]
LGTM

**Comment by kentcdodds** [comment]
@sxn, thanks so much for this feature! I've added you as a collaborator on the project. Accept that invitation and feel free to squash and merge this PR yourself :)

Feel free to watch the repo and continue making contributions as you feel the need/desire :) Thanks again!

**Comment by sxn** [comment]
Nice! Thanks, will do. :)

**Comment by kentcdodds** [comment]
Oh one sec! Could you add some docs in the README for this feature?

**Comment by sxn** [comment]
Sure thing!

**Comment by kentcdodds** [comment]
LGTM

**Pull request merged**